realm/rfc/0001-backlink-query-syntax.md

```diff
@@ -0,0 +1,204 @@
+- Feature Name: Backlink query syntax
+- Start Date: 2016-05-30
+- RFC PR: 
+
+# Summary
+
+It should be possible to define and query backlinks in both `Realm` and 
+`DynamicRealm`. As a `DynamicRealm` can work without _any_ Realm model classes,
+it should be possible to query backlinks without them being defined there.
+
+# Motivation
+
+Backlinks are special in Realm. They are not defined explicitly in the schema,
+but are automatically created when defining Links and LinkLists (forward links).
+
+This creates some conflicts with the current query system as it rely on all 
+fields being defined by the schema. This is called *named backlinks*.
+
+This proposal introduces a new syntax for queries so it is possible to query
+across a backlink even though it hasn't been defined by a model class. This is
+called *unnamed backlinks*.
+
+Any design should ideally solve all these constraints:
+
+- Should support a list of backlinks.
+
+- Should support a single backlink, enforcing the constraint that only 1 object 
+  can point to it. This is not supported by Core, but it is a natural extension.
```

Comment:
I think this will be problematic. From a high-level perspective there is nothing like this in object oriented programming. Objects can point to one or multiple other objects (based on the property types), but there is no way to limit who can point to any given object (it is just pointers).

That does not mean that we could not extend further than what is natural in oop (we already do that in several areas), but it will seem as a bit of a foreign concept.

The big question is how it should be enforced? Should it throw an exception if you try to point to an object that already has pointers to it? Or maybe null/remove the the previous link? Could end up seeming a bit "magic".